Source/WebCore/rendering/GridBaselineAlignment.cpp

```diff
@@ -123,31 +126,25 @@ void GridBaselineAlignment::updateBaselineAlignmentContext(ItemPosition preferen
 
     // Determine Ascent and Descent values of this child with respect to
     // its grid container.
-    LayoutUnit ascent = ascentForChild(child, baselineAxis);
-    if (isDescentBaselineForChild(child, baselineAxis))
-        ascent = descentForChild(child, ascent, baselineAxis);
-
+    LayoutUnit ascent = logicalAscentForChild(child, baselineAxis, preference);
     // Looking up for a shared alignment context perpendicular to the
     // baseline axis.
     bool isRowAxisContext = baselineAxis == GridColumnAxis;
     auto& contextsMap = isRowAxisContext ? m_rowAxisAlignmentContext : m_colAxisAlignmentContext;
-    auto addResult = contextsMap.add(sharedContext, nullptr);
-
+    auto* contextSearch = contextsMap.get(sharedContext);
     // Looking for a compatible baseline-sharing group.
-    if (addResult.isNewEntry)
-        addResult.iterator->value = makeUnique<BaselineContext>(child, preference, ascent);
-    else {
-        auto* context = addResult.iterator->value.get();
-        context->updateSharedGroup(child, preference, ascent);
-    }
+    if (!contextSearch)
```

Comment:
Super minor nit, would prefer the positive condition branch first, maybe `if (auto* contextSearch = ....)`

Reply:
Changed!